src/select/SelectMenu.ts

```diff
@@ -1,77 +1,56 @@
 /* eslint-disable react-hooks/exhaustive-deps */
 import React from "react";
-import { createOnKeyDown } from "reakit-utils";
+import { usePopover } from "reakit";
 import { BoxHTMLProps } from "reakit/ts/Box/Box";
-import { SelectStateReturn } from "./useSelectState";
-import { createHook, createComponent } from "reakit-system";
+import { SelectStateReturn } from "./SelectState";
 import { SELECT_KEYS } from "./__keys";
-
-export type SelectMenuOptions = Pick<
-  SelectStateReturn,
-  "setTypehead" | "selected" | "hide" | "show" | "visible"
-> & {
-  onChange?: (value: any) => void;
-};
+import { createComponent, createHook } from "reakit-system";
+import { CompositeProps, useComposite } from "reakit/Composite";
+import { usePortalShortcut } from "./__utils";
+import { useForkRef } from "reakit-utils";
+
+export type SelectMenuOptions = CompositeProps &
+  Pick<
+    SelectStateReturn,
+    "setSelected" | "visible" | "values" | "currentId" | "move"
+  > & {
+    maxHeight?: string | number;
+    modal?: boolean;
+  };
 
 const useSelectMenu = createHook<SelectMenuOptions, BoxHTMLProps>({
   name: "SelectMenu",
-  keys: SELECT_KEYS,
+  compose: [useComposite, usePopover],
+  keys: ["maxHeight", ...SELECT_KEYS],
+  useOptions({ modal = false, ...options }) {
+    return { modal, ...options };
+  },
```

Comment:
Any particular reason, why you have set `modal` to false.

No used inside `useProps` too. Just trying to understand.

Reply:
Seen it here, so i put that in, https://github.com/reakit/reakit/blob/master/packages/reakit/src/Popover/Popover.ts#L25

Reply:
If it's already there, then you don't have to put in, that will also be executed.
So no worries

Reply:
You can remove these.